### PR TITLE
Cosmetic Changes Search Functionality and Highlight, Comments UI, Save UI, DB calls

### PR DESCRIPTION
**Pull request by santheman12**

### CODE REVIEW

File: lib/calls.tsx [hunk 1]
```diff
@@ -12,23 +12,40 @@ export async function getAllExperiences() {
...
+    const highlightQuery = (text: string, query: string) => {
+        const highlightedText = text.replace(new RegExp(query, 'gi'), (match) => `<mark>${match}</mark>`);
+        return highlightedText;
+    };
+
     // filter per query
-    const filterdExperiences = allExperiences.data.filter((exp: any) => exp.description?.includes(query));
+    // const filterdExperiences = allExperiences.data.filter((exp: any) => (exp.description?.toLowerCase().includes(query) | exp.title?.toLowerCase().includes(query)));
+    const filteredExperiences = allExperiences.data.filter((exp: any) => {
+        const queryLower = query.toLowerCase();
```
Comment (crowdedpoem) [inline lib/calls.tsx:28]: I was wondering about this. I think there should be a file called `libs/normalize/ts` where we specify how we will trim whitespace and make text case insensitive before adding to the database

File: src/components/SearchServerParams.tsx [hunk 2]
```diff
@@ -1,62 +1,58 @@
...
 
-    // EFFECT: Set Initial Params
     useEffect(() => {
-        const params = new URLSearchParams(window.location.search)
-        const searchQuery = params.get("search") ?? ""
-        setInputValue(searchQuery)
-    }, [])
+        const params = new URLSearchParams(window.location.search);
+        const searchQuery = params.get("search") ?? "";
+        setInputValue(searchQuery.toLowerCase());
```
Comment (crowdedpoem) [inline src/components/SearchServerParams.tsx:34]: another place to use `libs/normalize.ts`

**Review by crowdedpoem — CHANGES_REQUESTED** [review]
Overall, really good. Just would like to see a `libs/normalization.ts` library function that normalizes input for the database. It would improve search reliability greatly. Also, maybe handle dates with built in libraries

**Comment by santheman12** [comment]
> Overall, really good. Just would like to see a `libs/normalization.ts` library function that normalizes input for the database. It would improve search reliability greatly. Also, maybe handle dates with built in libraries

will need to format the date on the postCard as well, but will wait for the merging to be complete for that

**Review by crowdedpoem — APPROVED** [review]

**Pull request merged**